ush/oceanice_nc2grib2.sh

```diff
@@ -0,0 +1,319 @@
+#!/bin/bash
+
+# This script contains functions to convert ocean/ice rectilinear netCDF files to grib2 format
+# This script uses the wgrib2 utility to convert the netCDF files to grib2 format and then indexes it
+
+source "${HOMEgfs}/ush/preamble.sh"
+
+################################################################################
+function _ice_nc2grib2 {
+# This function converts the ice rectilinear netCDF files to grib2 format
+
+  # Set the inputs
+  local grid=${1} # 0p25, 0p50, 1p00, 5p00
+  local latlon_dims=${2} # 0:721:0:1440, 0:361:0:720, 0:181:0:360, 0:36:0:72
+  local vdate=${3} # YYYYMMDDHH
+  local aperiod=${4} # 0-6
+  local infile=${5} # ice.0p25.nc
+  local outfile=${6} # ice.0p25.grib2
+  local template=${7} # template.global.0p25.gb2
+
+  ${WGRIB2} "${template}" \
+  -import_netcdf "${infile}" "hi_h" "0:1:${latlon_dims}" \
+      -set_var ICETK -set center 7 \
+      -set_date "${vdate}" -set_ftime "${aperiod} hour ave fcst" \
+      -set_scaling same same -set_grib_type c1 -grib_out "${outfile}" \
+  -import_netcdf "${infile}" "aice_h" "0:1:${latlon_dims}" \
+      -set_var ICEC -set center 7 \
+      -set_date "${vdate}" -set_ftime "${aperiod} hour ave fcst" \
+      -set_scaling same same -set_grib_type c1 -grib_out "${outfile}" \
+  -import_netcdf "${infile}" "Tsfc_h" "0:1:${latlon_dims}" \
+      -set_var ICETMP -set center 7 -rpn "273.15:+" \
+      -set_date "${vdate}" -set_ftime "${aperiod} hour ave fcst" \
+      -set_scaling same same -set_grib_type c1 -grib_out "${outfile}" \
+  -import_netcdf "${infile}" "uvel_h" "0:1:${latlon_dims}" \
+      -set_var UICE -set center 7 \
+      -set_date "${vdate}" -set_ftime "${aperiod} hour ave fcst" \
+      -set_scaling same same -set_grib_type c1 -grib_out "${outfile}" \
+  -import_netcdf "${infile}" "vvel_h" "0:1:${latlon_dims}" \
+      -set_var VICE -set center 7 \
+      -set_date "${vdate}" -set_ftime "${aperiod} hour ave fcst" \
+      -set_scaling same same -set_grib_type c1 -grib_out "${outfile}"
```

Comment:
This should be close, but feel free to save it for a follow-up PR.

Would love to make this a function, but need bash >=4.3 to do it the nice way (without needing to serialize/deserialize it ourselves).
```suggestion
  declare -A in_to_out
  in_to_out['hi_h']="ICETK"
  in_to_out['aice_h']="ICEC"
  in_to_out['Tsfc_h']="ICETMP"
  in_to_out['uvel_h']="UICE"
  in_to_out['vvel_h']="VICE"
  # These variables are required for GFS v17/GEFS v13, but are not
  #   yet available in NCEP GRiB2 tables as of wgrib v2.0.8
  # in_to_out['hs_h']="???"
  # in_to_out['frzmlt_h']="???"
  # in_to_out['albsni_h']="???"
  # in_to_out['mlt_onset_h']="???"
  # in_to_out['frz_onset_h']="???"

  local import_string=''
  local in_var out_var
  for in_var in ${!in_to_out[@]}; do
    out_var="${in_to_out[${in_var}]}"
    import_string="${import_string} -import_netcdf \"${infile}\" \"${in_var}\" \"0:1:${latlon_dims}\" -set_var \"${out_var}\" -set center 7 -set_date \"${vdate}\" -set_ftime \"${aperiod} hour ave fcst\" -set_scaling same same -set_grid_type c1 -grib_out \"${outfile}\""
  done
  
  # Do NOT quote ${import_string}
  ${WGRIB2} "${template}" ${import_string}
```

Reply:
will save it for a follow up PR.  Hopefully, we will generate the grib2 directly from the executable instead of wgrib2.